src/utils/errors.ts

```diff
@@ -20,26 +20,9 @@ export function mixin<T extends ErrorType>(this: unknown, ErrorClass: T): T {
   assert(this instanceof mixin === false, 'mixin(...) is not a class.');
   assert(typeof ErrorClass === 'function', 'Expected an error class.');
 
-  class RetreonErrorMixin extends ErrorClass {
+  return class RetreonErrorMixin extends ErrorClass {
     [KNOWN_ERROR_MARKER] = true;
-
-    constructor(...args: any[]) {
-      super(...args);
-
-      // DX improvement: try to assign a meaningful name (it shows up in the
-      // error description). If the mixin is used without a subclass, then
-      // don't override the name.
-      if (this.constructor.name !== 'RetreonErrorMixin') {
-        Object.defineProperty(RetreonErrorMixin.prototype, 'name', {
-          value: this.constructor.name || ErrorClass.name,
-          configurable: true,
-          writable: true,
-        });
-      }
-    }
```

Comment:
TypeScript doesn't assign a `constructor` field on the prototype as part of class inheritance, meaning this check breaks completely. It's caused too much trouble for what its worth.